modules/custom/entity_access_by_field/tests/src/Unit/EntityAccessTest.php

```diff
@@ -50,12 +48,34 @@ class EntityAccessTest extends UnitTestCase {
    */
   protected $nodeOwnerId;
 
+  /**
+   * The prophecy object.
+   *
+   * @var \Prophecy\Prophet
+   */
+  private $prophet;
+
+  /**
+   * Set up.
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->prophet = new Prophet();
+  }
+
+  /**
+   * Tear down.
+   */
+  protected function tearDown(): void {
+    $this->prophet->checkPredictions();
+  }
```

Comment:
Is it a good idea to do this in tear down rather than on a per-test basis? I'm not sure how reporting will differ ("There was an error in cleaning up" vs "Your test failed").

I.e. the underlying question is: Does PHPUnit already consider the test case a success if it performs tearDown? If the answer is yes, we should move this.

Reply:
Hi Alexander, 

> Is it a good idea to do this in tear down rather than on a per-test basis? I'm not sure how reporting will differ ("There was an error in cleaning up" vs "Your test failed").

According to https://phpunit.readthedocs.io/en/9.5/fixtures.html#fixtures `setUp()` and `tearDown()` template methods are run **_once for each test method_** (and on fresh instances) of the test case class. and Example Example 4.2 has a nice explanation on how they work.

> I.e. the underlying question is: Does PHPUnit already consider the test case a success if it performs tearDown? If the answer is yes, we should move this.

According to official documentation from prophecy `tearDown()` is where we should call `checkPredictions()`
Please check here for more information: https://github.com/phpspec/prophecy#predictions

Reply:
If official documentation says it, then that's good enough for me 👍